File: readability/readability.c
#include <stdio.h>

float coleman_liau(const char * text);

int main(void)
{
    char * text;
    do
    {
        text = get_string("Text: ");
    }
    while (text == NULL);

    int grade = (int) roundf(coleman_liau(text));

    if (grade > 16)
        printf("Grade 16+\n");
    else if (grade < 1)
        printf("Before grade 1\n");
    else
        printf("Grade %d\n", grade);

    return 0;
}

float coleman_liau(const char * text)
{
    int word_count = 0;
    int sentence_count = 0;
    
    return 0.0588 * ((float) L) - 0.296 * ((float) S) - 15.8;
}

File: sentimental-readability/readability.py
def main():
    text = input("Text: ")

    index = coleman_liau(text)
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


def coleman_liau(text):
    word_count = len(text.split())
    sentence_count = len(text.split("!?."))

    letter_count = 0
    for letter in text:
        if letter.isalpha():
            letter_count += 1

    avg_letters = letter_count / word_count * 100.0
    avg_sentences = sentence_count / word_count * 100.0

    print(word_count, sentence_count, letter_count, avg_letters, avg_sentences)

    return round(0.0588 * avg_letters - 0.296 * avg_sentences - 15.8)


main()
